### PR TITLE
adding kadane's algorithm in c

### LINKED ISSUE #287: Kadane's Algorithm
Write an efficient program to find the sum of contiguous subarray within a one-dimensional array of numbers which has the largest sum.

**Is your feature request related to a problem? Please describe.**
It is not a problem. It is an efficient algorithm to find largest subarray in sum in O(n) complexity

**Describe the solution you'd like**
Simple idea of the Kadane’s algorithm is to look for all positive contiguous segments of the array.

**Additional context**

**Languages**

-   [x] Python : @13snehitha
-   [x] Java : @DefCon-018
-   [x] C: @rgarg-2711
-   [x] C++: @Blackwidow2981

### PR DESCRIPTION
**Pull request by rgarg-2711**
# Description adding kadane's algorithm in c

Please include a summary of the changes you have made in regard with this issue. 

Fixes #287 

## Type of change

**Please delete options that are not relevant**

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [x] This change requires a documentation update

# Checklist:

-   [x] I have read the contribution guidelines before making this PR.
-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented in my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation (Language folder or any other readme instructed to)
-   [x] My changes generates no new warnings
-   [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by rgarg-2711** [comment]
in reference to issue #287

File: C/kadane.c [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    {
+        scanf("%d",&a[i]);
+    }
+
+    int max_sum = maxSubArraySum(a, n); 
+    printf("Maximum contiguous sum is %d",max_sum);
+
+    return 0; 
+} 
+/*run the test case
```
Comment (kaustubhgupta) [inline C/kadane.c]: Remove this doc string

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

**Comment by rgarg-2711** [comment]
in reference to issue #287 update the kadane's algo in c

**Comment by kaustubhgupta** [comment]
@YashSaxena9 Again strong sus of plagiarism :
See this: (Maybe I am wrong )
https://gist.github.com/sujaykundu777/a1a72c88291f9ee72aaba02b80159bf7

**Comment by rgarg-2711** [comment]
I did it on my personal IDE.It's the basic code find everywhere same.

**Review by kaustubhgupta — APPROVED** [review]

**Pull request merged**